resources/idlharness.js

```diff
@@ -3010,7 +3018,143 @@ function IdlTypedef(obj)
+//@}
+
+IdlNamespace.prototype.do_member_operation_asserts = function (memberHolderObject, member, a_test)
+//@{
+{
+    var done = a_test.done.bind(a_test);
+    var operationUnforgeable = member.has_extended_attribute("Unforgeable");
+    var desc = Object.getOwnPropertyDescriptor(memberHolderObject, member.name);
+
+    assert_false("get" in desc, "property should not have a getter");
+    assert_false("set" in desc, "property should not have a setter");
+    assert_equals(
+        desc.writable,
+        !operationUnforgeable,
+        "property should be writable if and only if not unforgeable");
+    assert_true(desc.enumerable, "property should be enumerable");
+    assert_equals(
+        desc.configurable,
+        !operationUnforgeable,
+        "property should be configurable if and only if not unforgeable");
+
+    assert_equals(
+        typeof memberHolderObject[member.name],
+        "function",
+         "property must be a function");
+
+    var args = member.arguments.map(a => create_suitable_object(a.idlType));
+    this.array.assert_type_is(
+        memberHolderObject[member.name].call(null, ...args),
+        member.idlType);
+
+    assert_equals(memberHolderObject[member.name].length,
+        minOverloadLength(this.members.filter(function(m) {
+            return m.type == "operation" && m.name == member.name;
+        })),
+        "operation has wrong .length");
+    done();
+}
+//@}
+
+IdlNamespace.prototype.test_member_operation = function(member)
+//@{
+{
+    if (!shouldRunSubTest(this.name)) {
+        return;
+    }
+    const args = member.arguments
+        .map(a => `${a.idlType.idlType}${a.variadic ? '...' : ''}`)
+        .join(", ");
+    var a_test = subsetTestByKey(
+        this.name,
+        async_test,
+        `${this.name} namespace: operation ${member.name}(${args})`);
+    a_test.step(() => {
+        assert_own_property(
+            self[this.name],
+            member.name,
+            `namespace object missing operation ${format_value(member.name)}`);
+
+        this.do_member_operation_asserts(self[this.name], member, a_test);
+    });
+};
+//@}
+
+IdlNamespace.prototype.test_member_attribute = function (member)
+//@{
+{
+    if (!shouldRunSubTest(this.name)) {
+        return;
+    }
+    var a_test = subsetTestByKey(
+        this.name,
+        async_test,
+        `${this.name} namespace: attribute ${member.name}`);
+    a_test.step(function()
+    {
+        assert_own_property(
+            self[this.name],
+            member.name,
+            `${this.name} does not have property ${format_value(member.name)}`);
+
+        var desc = Object.getOwnPropertyDescriptor(self[this.name], member.name);
+        // "The attribute setter is undefined if the attribute is declared
+        // readonly and has neither a [PutForwards] nor a [Replaceable]
+        // extended attribute declared on it."
+        assert_equals(desc.set, undefined, "setter must be undefined for readonly attributes");
```

Comment:
This unconditional check does not seem to match the quoted spec text.

Reply:
Conditions added.